File: backend/models/sticky_note.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type StickyNote struct {
	ID        string         `json:"id"         gorm:"primaryKey;type:text"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-"          gorm:"index"`
	ProjectID string         `json:"project_id" gorm:"not null;index;type:text"`
	Content   string         `json:"content"`
	PosX      float64        `json:"pos_x"`
	PosY      float64        `json:"pos_y"`
	Color     string         `json:"color"`
	CreatedBy string         `json:"created_by" gorm:"type:text"`
	Author    string         `json:"author"`
	ParentID    *string        `json:"parent_id"   gorm:"type:text"`
	IsGroup     bool           `json:"is_group"`
	MatrixCost  *float64       `json:"matrix_cost"`
	MatrixValue *float64       `json:"matrix_value"`
}

func (n *StickyNote) BeforeCreate(tx *gorm.DB) error {
	if n.ID == "" {
		n.ID = uuid.New().String()
	}
	return nil
}

File: frontend/lib/models/sticky_note.dart
class StickyNote {
  final String id;
  final String content;
  final double posX;
  final double posY;
  final String color;
  final String author;
  final String createdBy;
  final String? parentId;
  final bool isGroup;
  final double? matrixCost;
  final double? matrixValue;

  const StickyNote({
    required this.id,
    required this.content,
    required this.posX,
    required this.posY,
    required this.color,
    required this.author,
    required this.createdBy,
    this.parentId,
    this.isGroup = false,
    this.matrixCost,
    this.matrixValue,
  });

  factory StickyNote.fromJson(Map<String, dynamic> json) => StickyNote(
        id: json['id'] as String,
        content: json['content'] as String? ?? '',
        posX: (json['pos_x'] as num).toDouble(),
        posY: (json['pos_y'] as num).toDouble(),
        color: json['color'] as String? ?? '#FFF176',
        author: json['author'] as String? ?? '',
        createdBy: json['created_by'] as String? ?? '',
        parentId: json['parent_id'] as String?,
        isGroup: json['is_group'] as bool? ?? false,
        matrixCost: (json['matrix_cost'] as num?)?.toDouble(),
        matrixValue: (json['matrix_value'] as num?)?.toDouble(),
      );

  StickyNote copyWith({
    String? content,
    double? posX,
    double? posY,
    String? parentId,
    bool clearParent = false,
  }) =>
      StickyNote(
        id: id,
        content: content ?? this.content,
        posX: posX ?? this.posX,
        posY: posY ?? this.posY,
        color: color,
        author: author,
        createdBy: createdBy,
        parentId: clearParent ? null : (parentId ?? this.parentId),
        isGroup: isGroup,
      );
}
